pkg/types/aws/validation/machinepool.go

```diff
@@ -54,6 +58,40 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 
 	allErrs = append(allErrs, validateSecurityGroups(platform, p, fldPath)...)
 
+	if p.HostPlacement != nil {
+		allErrs = append(allErrs, validateHostPlacement(p, fldPath.Child("hostPlacement"))...)
+	}
+
+	return allErrs
+}
+
+func validateHostPlacement(p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.HostPlacement.Affinity != nil {
+		switch *p.HostPlacement.Affinity {
+		case aws.HostAffinityAnyAvailable:
+			if p.HostPlacement.DedicatedHost != nil {
+				allErrs = append(allErrs, field.Required(fldPath.Child("dedicatedHost"), "dedicatedHost is required when 'affinity' is set to DedicatedHost, and forbidden otherwise"))
+			}
+		case aws.HostAffinityDedicatedHost:
+			if p.HostPlacement.DedicatedHost == nil {
+				allErrs = append(allErrs, field.Required(fldPath.Child("dedicatedHost"), "dedicatedHost is required when 'affinity' is set to DedicatedHost, and forbidden otherwise"))
+			} else {
+				for index, host := range *p.HostPlacement.DedicatedHost {
+					hostPath := fldPath.Child("dedicatedHost").Index(index)
+					if len(host.ID) == 0 {
+						allErrs = append(allErrs, field.Required(hostPath.Child("id"), "a hostID must be specified when configuring 'dedicatedHost'"))
+					} else if awsDedicatedHostNamePattern.FindStringSubmatch(host.ID) == nil {
+						allErrs = append(allErrs, field.Invalid(hostPath.Child("id"), host.ID, "id must start with 'h-' followed by 17 lowercase hexadecimal characters (0-9 and a-f)"))
+					}
+				}
+			}
+		default:
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("affinity"), p.HostPlacement.Affinity, "affinity must be either AnyAvailable or DedicatedHost"))
```

Comment:
```suggestion
			allErrs = append(allErrs, field.NotSupported(fldPath.Child("affinity"), p.HostPlacement.Affinity, []aws.HostAffinity{aws.HostAffinityAnyAvailable, aws.HostAffinityDedicatedHost}))
```

nit: We can use `field.Unsupported` func.